documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -1715,6 +1715,33 @@ declarative <<writing-tests-parallel-execution-synchronization, synchronization>
 mechanism. Please note that <<running-tests-capturing-output>> needs to be enabled
 separately.
 
+In addition, you can configure the default execution mode for top-level classes by setting
+the `junit.jupiter.execution.parallel.mode.classes.default` configuration parameter. By
+combining both configuration parameters, you can configure classes to run in parallel but
+their methods in the same thread:
+
+[source,properties]
+.Configuration parameters to execute top-level classes in parallel but methods in same thread
+----
+junit.jupiter.execution.parallel.enabled = true
+junit.jupiter.execution.parallel.mode.default = same_thread
+junit.jupiter.execution.parallel.mode.classes.default = concurrent
+----
+
+The opposite combination will run all methods within one class in parallel, but top-level
+classes will run sequentially:
+
+[source,properties]
+.Configuration parameters to execute top-level classes in sequentially but their methods in parallel
+----
+junit.jupiter.execution.parallel.enabled = true
+junit.jupiter.execution.parallel.mode.default = concurrent
+junit.jupiter.execution.parallel.mode.classes.default = same_thread
+----
+
+Not configuring a value for `junit.jupiter.execution.parallel.mode.classes.default` is
+equivalent to using the same value as for `junit.jupiter.execution.parallel.mode.default`.
+
```

Comment:
Add a paragraph showing both _default_ keys set to `concurrent`:

```
----
junit.jupiter.execution.parallel.enabled = true
junit.jupiter.execution.parallel.mode.default = concurrent
junit.jupiter.execution.parallel.mode.classes.default = concurrent
----
```

Reply:
That's the same as the following so I didn't think it made sense to include it. WDYT?

```
----
junit.jupiter.execution.parallel.enabled = true
junit.jupiter.execution.parallel.mode.default = concurrent
----
```

Reply:
Yeah, might feel like an unnecessary repetition, but it doesn't hurt to be explicit here.

_Not configuring a value for [...] is equivalent to using the same value as [...]._
_It unrolls to this configuration "2x concurrent", which means classes and methods are executed in any order._

Reply:
Perhaps we also should _paint out_ a potential execution order example like this?

- same, same: `T{a1, a2, a3, b1, b2, b3, c1, c2}`
- same, concurrent: `TA{a1, a3, a2}, TB{b3, b2, b1}, TC{c2, c1}`
- concurrent, same: `TA{a1, a2, a3}, TB{b1, b2, b3}, TC{c1, c2}`
- concurrent, concurrent: `TX{a1, b2, c2, b3}, TY{b1, a3}, TZ{c1, a2}`

Reply:
A series of gantt-like charts would be a good illustration. 🤔

Reply:
### same, same
```
T -> a1 - a2 - b1 - b2
```

### same, concurrent
```
T1 -> a1 - a2
T2 -> b1 - b2
```

### concurrent, same
```
T1 -> a1 - b1
T2 -> a2 - b2
```

### concurrent, concurrent
```
T1 -> a1
T2 -> a2
T3 -> b1
T4 -> b2
```

Correct?

Reply:
Depends on what `a1` and so on mean. 😉

Reply:
I'll go ahead, merge this PR and will create a follow-up issue to improve the documentation.